web/indexProm.js

```diff
@@ -0,0 +1,68 @@
+// var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+const API = 'https://rickandmortyapi.com/api/character/';
+const $image = document.querySelector("#image")
+const $name = document.querySelector("#name")
+const $dimension = document.querySelector("#dimension")
+
+const fetchData = (urlApi) => {
+  const xhttp = new XMLHttpRequest();
+
+  return new Promise((resolve, reject) => {
+    xhttp.onreadystatechange =  () => { 
+        if(xhttp.readyState === 4) {
+          if(xhttp.status === 200){
+            return resolve(xhttp.responseText);
+          } else {
+            return reject(xhttp.statusText)
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```